File: backend/internal/navigation/model/navigation.go
package model

type NavigationItem struct {
	ID        string
	Code      string
	Title     string
	Icon      string
	ParentID  *string
	Component string
	SortOrder int
	CanView   bool
	CanCreate bool
	CanUpdate bool
	CanDelete bool
	IsDefault bool
}

File: frontend/src/types/navigation.ts
export type NavigationItem = {
  id: string;
  code: string;

  title: string;
  icon: string;

  parent_id?: string | null;

  component?: string | null;

  sort_order?: number;

  is_active?: boolean;

  is_default?: boolean;

  children?: NavigationItem[];
};
